x/oracle/keeper/historic_avg_test.go

```diff
@@ -1,272 +1,267 @@
+// 	for i := 1; i < int(numCounters); i++ {
+// 		checkCounter(t, k, s.denom1, byte(i), "3", 1)
+// 	}
+
+// 	// cycle over by 2 shifts -> going to index=1
+// 	// after the cycle, the current index should be the one above it -> 2
+// 	now = now.Add(shift * 2)
+// 	k.updateAvgCounter(s.denom1, sdk.NewDec(2), now)
+// 	checkCounter(t, k, s.denom1, 0, "2", 1)
+// 	checkCounter(t, k, s.denom1, 1, "2", 1)
+// 	checkCounter(t, k, s.denom1, 2, "5", 2) // num=2 because the very first update was only registered for counter[0]
+// 	for i := 3; i < int(numCounters); i++ {
+// 		checkCounter(t, k, s.denom1, byte(i), "5", 2)
+// 	}
+// 	checkAvgPrice(t, k, "2.5", s.denom1, 2)
+
+// 	now = now.Add(time.Minute)
+// 	k.updateAvgCounter(s.denom1, sdk.NewDec(1), now)
+// 	checkAvgPrice(t, k, "2", s.denom1, 2)
+
+// 	now = now.Add(k.shift)
+// 	k.updateAvgCounter(s.denom1, sdk.NewDec(2), now)
+// 	checkAvgPrice(t, k, "2", s.denom1, 3)
+// 	for i := 3; i < int(numCounters); i++ {
+// 		checkCounter(t, k, s.denom1, byte(i), "8", 4)
+// 	}
+// 	checkCounter(t, k, s.denom1, 2, "2", 1)
+// 	checkCounter(t, k, s.denom1, 1, "5", 3)
+// 	checkCounter(t, k, s.denom1, 0, "5", 3)
+// }
+
+// func (s AvgKeeperSuite) testUpdateAvgCounterHalt(t *testing.T) {
+// 	now, shift, k := s.setupUpdateAvgCounter(t)
+// 	k.updateAvgCounter(s.denom1, sdk.NewDec(1), now)
+// 	checkAvgPrice(t, k, "1", s.denom1, 0)
+
+// 	var numCounters = k.numCounters()
+// 	// go to the latest shift in the epoch
+// 	now = now.Add(shift * time.Duration(numCounters-1))
+// 	k.updateAvgCounter(s.denom1, sdk.NewDec(3), now)
+// 	checkAvgPrice(t, k, "2", s.denom1, 0)
+
+// 	// go 2 periods forward
+// 	now = now.Add(shift * time.Duration(numCounters))
+// 	k.updateAvgCounter(s.denom1, sdk.NewDec(6), now)
+// 	checkAvgPrice(t, k, "6", s.denom1, 0)
+
+// 	// go 2 periods -1 shift forward
+// 	now = now.Add(shift * time.Duration(2*numCounters-1))
+// 	k.updateAvgCounter(s.denom1, sdk.NewDec(7), now)
+// 	checkAvgPrice(t, k, "7", s.denom1, byte(numCounters)-1)
+// }
+
+// func checkAvgPrice(t *testing.T, k AvgKeeper, expected, denom string, idx byte) {
+// 	i, err := k.getLatestIdx(denom)
+// 	if err != types.ErrNoLatestAvgPrice {
+// 		require.NoError(t, err)
+// 	}
+// 	require.Equal(t, idx, i, "index check")
+
+// 	expectedDec := sdk.MustNewDecFromStr(expected)
+// 	v, err := k.GetCurrentAvg(denom)
+// 	require.NoError(t, err)
+// 	require.Equal(t, expectedDec, v) // ussing testify to have a stack trace
+// }
+
+// func checkCounter(t *testing.T, k AvgKeeper, denom string, idx byte, sum string, num uint32) {
+// 	sumDec := sdk.MustNewDecFromStr(sum)
+// 	c, err := k.getCounter(denom, idx)
+// 	require.NoError(t, err)
+// 	tassert.Equal(t, sumDec, c.Sum, "sum idx=%d", idx)
+// 	tassert.Equal(t, num, c.Num, "num idx=%d", idx)
+// }
+
+// func newAvgCounter(sum, num uint32, start time.Time) types.AvgCounter {
+// 	return types.AvgCounter{Sum: sdk.NewDec(int64(sum)), Num: num, Start: start}
+// }
```

Comment:
why this is commented out?

Reply:
I added previously to test the codecov, I will remove it